packages/artifact/src/internal/download-http-client.ts

```diff
@@ -263,26 +263,47 @@ export class DownloadHttpClient {
       if (isGzip) {
         const gunzip = zlib.createGunzip()
         response.message
+          .on('error', error => {
+            core.error(
+              `An error occurred while attempting to read the response stream`
+            )
+            reject(error)
+            gunzip.close()
+          })
           .pipe(gunzip)
+          .on('error', error => {
```

Comment:
Reviewers: The existing `on('error', ...)` callback below isn't called when gunzip fails. `pipe` doesn't propagate errors across streams, so you have to monitor each.

Without this block the error is unhandled, and the new test times out.